src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala

```diff
@@ -354,6 +348,44 @@ class ClusterMonitorActor(val cluster: Cluster,
     }.void
   }
 
+  private def saveClusterError(errorMessage: String, errorCode: Int): Future[Unit] = {
+    dbRef.inTransaction { dataAccess =>
+      val clusterId = dataAccess.clusterQuery.getIdByUniqueKey(cluster)
+      Future(logger.info(s"${cluster.clusterName}: clusterId in else is ${clusterId}"))
+      clusterId flatMap {
+        case Some(a) => dataAccess.clusterErrorQuery.save(a, ClusterError(errorMessage, errorCode, Instant.now))
+        case None => {
+          logger.warn(s"Could not find Id for Cluster ${cluster.projectNameString}  with google cluster ID ${cluster.dataprocInfo.googleId}.")
+          DBIOAction.successful(0)
+        }
+      }
+    }.void.handleError(e => logger.info(s"Error persisting cluster error with message '${errorMessage}' to database: ${e}"))
```

Comment:
I'll use `handleError` on `persistClusterErrors` because you don't want to fail everything else because of failing in `persistClusterErrors`. With current implementation, if `google2StorageDAO.getObjectMetadata` fails, everything will fail, which isn't what you want